hal/network/lwip/esp32/esp32ncpnetif.cpp

```diff
@@ -204,28 +223,21 @@ void Esp32NcpNetif::loop(void* arg) {
 
 int Esp32NcpNetif::up() {
     NetifEvent ev = NetifEvent::Up;
-    int r = os_queue_put(queue_, &ev, 0, nullptr);
-    if (r) {
-        // FIXME: we'll miss this event, just as an attempt to potentially recover
-        // set internal state.
-        up_ = true;
-    }
-    return r;
+    return os_queue_put(queue_, &ev, 0, nullptr);
 }
 
 int Esp32NcpNetif::down() {
     const auto client = wifiMan_->ncpClient();
-    if (client->connectionState() != NcpConnectionState::CONNECTED) {
+    if (client->connectionState() != NcpConnectionState::CONNECTED && client->ncpState() == NcpState::ON) {
         // Disable the client to interrupt its current operation
         client->disable();
     }
     NetifEvent ev = NetifEvent::Down;
     int r = os_queue_put(queue_, &ev, 0, nullptr);
     if (r) {
-        // FIXME: we'll miss this event, just as an attempt to potentially recover
-        // set internal state.
-        up_ = false;
-        client->disable();
+        if (client->ncpState() == NcpState::ON) {
```

Comment:
We've just removed a workaround for a case where we for fail to post an event into the queue. We should still set the intended state here and call `disable()` to unblock the thread that processes these events. Otherwise it will be lost.

Reply:
Could elaborate how the previous fix works? If the queue is filled up, the next event popped from the queue will probably overwrite the `up_` flag, and the request is going to be lost as well.

Reply:
That's exactly why there is a FIXME there. Since we are refactoring the eventing, we should watch out for this. It's always possible to fill up the queue e.g. just issuing something in a loop accidentally.

```c++
void loop() {
    if (!Cellular.ready()) {
         Cellular.connect();
    } else if (Time.now() == 123123123) {
         // Just an example that this has a potential to get lost
         Cellular.off();
    }
}
```

We either have to live with the fact that we might need to block, or we can do something similar and e.g. calculate and store the intended state at the time of posting an event to the queue and be able to detect any discrepancies.

Reply:
I think we can go in the way that blocking with a certain timeout posting an event and discard the event that is the same  with the last posted event.

Reply:
That works as well, yeah.